src/repository/submission.js

```diff
@@ -86,3 +86,29 @@ export const getSubmissionsByQuestion = () => {
 export const getSubmissionCount = async (questionId) => {
   return (await Submission.distinct('user', { question: questionId })).length;
 };
+
+export const getLeaderboardData = () => {
+  return Submission.aggregate([
+    {
+      $group: {
+        _id: '$user',
+        name: { $first: '$user.name' },
+        email: { $first: '$user.email' },
+        total_score: { $sum: '$score' }
```

Comment:
Only the latest submission per user must be considered. The same has to be done in the get all users endpoint relative to the #106 issue. I'll work on it and you can use the same afterwards

Reply:
This is done, i've added the pipeline stages required to get the score over here:- 
https://github.com/sliit-foss/bashaway-backend/blob/a54953a6335c02a69c575a402bf656bcc2a412c6/src/repository/user/utils.js#L1
We could reuse the same. Also let's add the repository function to get the leaderboard info under users since it's base should be there. We don't need a service layer for this let's invoke the repository function directly from the controller